### PR TITLE
GS: Replace our own format iostream with using fmt lib

### PR DESCRIPTION
**Pull request by lightningterror**
### Description of Changes

Removes/replaces our own format iostream with fmt lib.
### Rationale behind Changes

Cleaner, less complex.
### Suggested Testing Steps

Test if HW/SW renderers are working properly.

### CODE REVIEW

File: pcsx2/GS/GSCrc.cpp [hunk 1]
```diff
@@ -17,6 +17,8 @@
 #include "GS.h"
 #include "GSCrc.h"
 
+#include "fmt/core.h"
```
Comment (tadanokojin) [inline pcsx2/GS/GSCrc.cpp:20]: use `#include <...>` for externals/system includes

File: pcsx2/GS/GSCapture.h [hunk 2]
```diff
@@ -20,6 +20,8 @@
 
 #ifdef _WIN32
 #include "Window/GSCaptureDlg.h"
+#else
+#include "fmt/core.h"
```
Comment (tadanokojin) [inline pcsx2/GS/GSCapture.h:24]: just include normally, it's not going to fail to compile or anything.

Comment (lightningterror) [reply pcsx2/GS/GSCapture.h:24]: I didn't because it's not used on windows here.

Comment (tadanokojin) [reply pcsx2/GS/GSCapture.h:24]: I understand. Just include it anyway.

File: pcsx2/GS/GSCrc.cpp (hunk 1 above)
Comment (lightningterror) [reply pcsx2/GS/GSCrc.cpp:20]: It's a 3rdparty submodule, won't compile on linux/mac otherwise.

**Review by F0bes — APPROVED** [review]
HW and SW modes seem to work fine on x64 linux.
Tested with both homebrew and Colin McRae Rally 3.

**Review by RedDevilus — APPROVED** [review]
SW/HW renderers work fine on Windows, so does the internal video recorder. Does need a final rebase :)

**Pull request closed without merging**